### PR TITLE
AMDGPU: Move attributor into optimization pipeline

### PR DESCRIPTION
**Pull request by arsenm**
Removing it from the codegen pipeline induces a lot of test churn because llc is no longer optimizing out implicit arguments to kernels.

Mostly mechanical, but there are some creative test updates. I preferred to take the changes as-is in tests where the ABI isn't relevant. In cases where it's more relevant, or the optimize out logic was too ingrained in the test, I pre-run the optimization. Some cases manually add attributes to disable inputs.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/SIFrameLowering.cpp [hunk 1]
```diff
@@ -678,6 +678,12 @@ void SIFrameLowering::emitEntryFunctionPrologue(MachineFunction &MF,
         break;
       }
     }
+
+    // FIXME: We can spill incoming arguments and restore at the end of the
+    // prolog.
+    if (!ScratchWaveOffsetReg)
```
Comment (shiltian) [inline llvm/lib/Target/AMDGPU/SIFrameLowering.cpp:684]: Is this change related?

**Review by shiltian — APPROVED** [review]
The non-test related changes look fairly straightforward. LGTM.

File: llvm/lib/Target/AMDGPU/SIFrameLowering.cpp (hunk 1 above)
Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIFrameLowering.cpp:684]: Kind of. One test hit the assert due to the sgpr limiting attribute

**Comment by rampitec** [comment]
The main question, is it required for correctness now or not? I.e. can we use -O0 w/o the attributor?

**Comment by arsenm** [comment]
> The main question, is it required for correctness now or not? I.e. can we use -O0 w/o the attributor?

No, it never was. The "required" stuff was in the other pass which was deleted already

**Review by rampitec — APPROVED** [review]

**Comment by arsenm** [comment]
This is stuck due to OpenCL blender failing in PSDB

**Comment by arsenm** [comment]
Latest attempt to pass testing worked. I'll re-resolve on all downstream branches and try submitting again next week

**Comment by dyung** [comment]
@arsenm are you aware that there is a test failure from this change that is still failing now about 16 hours later? Can you please take a look and revert if you need time to investigate so that we can get the bots back to green?

Some failing bots:
1. https://lab.llvm.org/buildbot/#/builders/144/builds/2221
2. https://lab.llvm.org/buildbot/#/builders/46/builds/1587
3. https://lab.llvm.org/buildbot/#/builders/190/builds/1798

**Comment by petrhosek** [comment]
I think we should go ahead and revert this change given that it's been almost an entire day and the builders are still broken.

**Comment by jplehr** [comment]
Hey @arsenm this broke all AMDGPU OpenMP Offload buildbots (e.g., https://lab.llvm.org/buildbot/#/builders/30).

Any chance you can fix these issues?

**Comment by arsenm** [comment]
> Hey @arsenm this broke all AMDGPU OpenMP Offload buildbots (e.g., https://lab.llvm.org/buildbot/#/builders/30).
> 
> Any chance you can fix these issues?

Can you attach before/after IR and just XFAIL them for now?
